app/jobs/bigquery_report_export_job.rb

```diff
@@ -0,0 +1,44 @@
+# frozen_string_literal: true
+
+require "google/cloud/bigquery"
+
+class BigqueryReportExportJob < ApplicationJob
+  DATASET_NAME = "daily"
+  ACCOUNTS_TABLE_NAME = "accounts"
+  EVENTS_TABLE_NAME = "events"
+
+  queue_as :default
+
+  def perform(start_date, end_date)
+    bigquery = Google::Cloud::Bigquery.new(credentials: Rails.application.secrets.bigquery_credentials)
+    dataset = bigquery.dataset DATASET_NAME
+
+    replace_accounts_table(dataset: dataset)
+    update_events_table(dataset: dataset, start_date: start_date, end_date: end_date)
+  end
+
+protected
+
+  def replace_accounts_table(dataset:)
+    report = Report::Accounts.report(
+      user_id_pepper: Rails.application.secrets.reporting_user_id_pepper,
+    )
+
+    delete_job = dataset.query_job "DELETE * FROM #{ACCOUNTS_TABLE_NAME}"
```

Comment:
Interesting, this is good because it should mean that if a user deletes their account their anon data disappears from the big query db right too?

It's a wholesale replace on each update?

Is this what PA / Data folks expect though? In this case some numbers that are otherwise cumulative might go down as well as up?

Reply:
So, the account data is replaced (as is the transition checker data in https://github.com/alphagov/govuk-attribute-service-prototype/pull/96), but the login data is incremental.

But when we implement the data retention of security events, we can mirror that here with something like `DELETE * FROM #{EVENTS_TABOE_NAME} WHERE `login_timestamp` < ...`